sentry/src/test/java/io/sentry/BaggageTest.kt

```diff
@@ -0,0 +1,178 @@
+        val baggage = Baggage.fromHeader(
+            listOf(
+                "userId =   alice",
+                "serverNode = DF%2028, isProduction = false"
+            ),
+            logger
+        )
+
+        assertEquals("alice", baggage.get("userId"))
+        assertEquals("DF 28", baggage.get("serverNode"))
+        assertEquals("false", baggage.get("isProduction"))
+
+        assertEquals("isProduction=false,serverNode=DF%2028,userId=alice", baggage.toHeaderString())
+    }
+
+    @Test
+    fun `can parse null baggage string`() {
+        val nothing: String? = null
+        val baggage = Baggage.fromHeader(nothing, logger)
+        assertEquals("", baggage.toHeaderString())
+    }
+
+    @Test
+    fun `can parse blank baggage string`() {
+        val baggage = Baggage.fromHeader("", logger)
+        assertEquals("", baggage.toHeaderString())
+    }
+
+    @Test
+    fun `can parse whitespace only baggage string`() {
+        val baggage = Baggage.fromHeader("   ", logger)
+        assertEquals("", baggage.toHeaderString())
+    }
+
+    @Test
+    fun `can parse whitespace only baggage strings`() {
+        val baggage = Baggage.fromHeader(listOf("   ", "   "), logger)
+        assertEquals("", baggage.toHeaderString())
+    }
+
+    @Test
+    fun `if value exceeds size limit key value pair is not added to header`() {
+        val largeValue = Faker.instance().random().hex(8193)
+        val baggage = Baggage.fromHeader("smallValue=remains,largeValue=$largeValue", logger)
+
+        assertEquals("remains", baggage.get("smallValue"))
+        assertNotNull(baggage.get("largeValue"))
+
+        assertEquals("smallValue=remains", baggage.toHeaderString())
+    }
+
+    @Test
+    fun `null value is omitted from header string`() {
+        val baggage = Baggage(logger)
+
+        baggage.setTraceId(null)
+        baggage.setPublicKey(null)
+        baggage.setRelease(null)
+        baggage.setEnvironment(null)
+        baggage.setTransaction(null)
+        baggage.setUserId(null)
+        baggage.setUserSegment(null)
+
+        assertEquals("", baggage.toHeaderString())
+    }
+
+    @Test
+    fun `can set values from trace context`() {
+        val baggage = Baggage(logger)
+        val publicKey = Dsn(dsnString).publicKey
+        val traceId = SentryId().toString()
+        val userId = UUID.randomUUID().toString()
+
+        baggage.setTraceId(traceId)
+        baggage.setPublicKey(publicKey)
+        baggage.setRelease("1.0-rc.1")
+        baggage.setEnvironment("production")
+        baggage.setTransaction("TX")
+        baggage.setUserId(userId)
+        baggage.setUserSegment("segmentA")
+
+        assertEquals("sentry-environment=production,sentry-publickey=$publicKey,sentry-release=1.0-rc.1,sentry-traceid=$traceId,sentry-transaction=TX,sentry-userid=$userId,sentry-usersegment=segmentA", baggage.toHeaderString())
+    }
+
+    @Test
+    fun `duplicate entries are lost`() {
```

Comment:
How do we want to handle duplicate entries?

Reply:
Let's stick with the test behaviour.

https://www.w3.org/TR/baggage/#mutating-baggage

> Because baggage entry keys, values, and metadata are not specified here, producers and consumers MAY agree on any set of mutation rules that don't violate the specification. For example, keys may be deduplicated by keeping the first entry, keeping the last entry, or concatenating values together.

We can revisit this if needed.